src/vm/hints/set.rs

```diff
@@ -0,0 +1,460 @@
+        elm_b: Option<&MaybeRelocatable>,
+    ) -> (VirtualMachine, HashMap<String, BigInt>) {
+        let mut vm = VirtualMachine::new(
+            BigInt::new(Sign::Plus, vec![1, 0, 0, 0, 0, 0, 17, 134217728]),
+            vec![(
+                "range_check".to_string(),
+                Box::new(RangeCheckBuiltinRunner::new(true, bigint!(8), 8)),
+            )],
+            false,
+        );
+
+        for _ in 0..3 {
+            vm.segments.add(&mut vm.memory, None);
+        }
+
+        vm.run_context.fp = MaybeRelocatable::from((0, 5));
+
+        let set_ptr_default = MaybeRelocatable::from((1, 0));
+        let elm_size_default = MaybeRelocatable::from(bigint!(2));
+        let elm_a_default = MaybeRelocatable::from(bigint!(2));
+        let elm_b_default = MaybeRelocatable::from(bigint!(3));
+
+        vm.memory
+            .insert(
+                &MaybeRelocatable::from((0, 2)),
+                if let Some(rel) = set_ptr {
+                    rel
+                } else {
+                    &set_ptr_default
+                },
+            )
+            .expect("Unexpected memory insert fail");
+        vm.memory
+            .insert(
+                &MaybeRelocatable::from((0, 3)),
+                if let Some(rel) = elm_size {
+                    rel
+                } else {
+                    &elm_size_default
+                },
+            )
+            .expect("Unexpected memory insert fail");
+        vm.memory
+            .insert(
+                &MaybeRelocatable::from((0, 4)),
+                &MaybeRelocatable::from((2, 0)),
+            )
+            .expect("Unexpected memory insert fail");
+        vm.memory
+            .insert(
+                &MaybeRelocatable::from((0, 5)),
+                &MaybeRelocatable::from((1, 2)),
+            )
+            .expect("Unexpected memory insert fail");
+        vm.memory
+            .insert(
+                &MaybeRelocatable::from((1, 0)),
+                &MaybeRelocatable::from(bigint!(1)),
+            )
+            .expect("Unexpected memory insert fail");
+        vm.memory
+            .insert(
+                &MaybeRelocatable::from((1, 1)),
+                &MaybeRelocatable::from(bigint!(3)),
+            )
+            .expect("Unexpected memory insert fail");
+        vm.memory
+            .insert(
+                &MaybeRelocatable::from((1, 2)),
+                &MaybeRelocatable::from(bigint!(5)),
+            )
+            .expect("Unexpected memory insert fail");
+        vm.memory
+            .insert(
+                &MaybeRelocatable::from((1, 3)),
+                &MaybeRelocatable::from(bigint!(7)),
+            )
+            .expect("Unexpected memory insert fail");
+        vm.memory
+            .insert(
+                &MaybeRelocatable::from((2, 0)),
+                if let Some(rel) = elm_a {
+                    rel
+                } else {
+                    &elm_a_default
+                },
```

Comment:
You can use https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap_or

Reply:
resolve: https://github.com/lambdaclass/cleopatra_cairo/pull/289/commits/0f008d1c33335380eb9dc3b7f52a9aededd12905